OpenRA.Mods.Common/Traits/Player/ProductionQueue.cs

```diff
@@ -160,6 +163,7 @@ void INotifyOwnerChanged.OnOwnerChanged(Actor self, Player oldOwner, Player newO
 			playerPower = newOwner.PlayerActor.TraitOrDefault<PowerManager>();
 			playerResources = newOwner.PlayerActor.Trait<PlayerResources>();
 			developerMode = newOwner.PlayerActor.Trait<DeveloperMode>();
+			techTree = newOwner.PlayerActor.Trait<TechTree>();
```

Comment:
Please use this cached version for the `.Update()` call below and in `CacheProducibles`.